builder/digitalocean/step_snapshot.go

```diff
@@ -60,6 +60,47 @@ func (s *stepSnapshot) Run(state multistep.StateBag) multistep.StepAction {
 		return multistep.ActionHalt
 	}
 
+	if len(c.SnapshotRegions) > 0 {
+		regionSet := make(map[string]struct{})
+		regions := make([]string, 0, len(c.SnapshotRegions))
+		regionSet[c.Region] = struct{}{}
+		for _, region := range c.SnapshotRegions {
+			// If we already saw the region, then don't look again
+			if _, ok := regionSet[region]; ok {
+				continue
+			}
+
+			// Mark that we saw the region
+			regionSet[region] = struct{}{}
+
+			regions = append(regions, region)
+		}
+		c.SnapshotRegions = regions
```

Comment:
can we make this a local variable? I think modifying the global config in a step could lead to unintended consequences